packages/editor/src/store/actions.js

```diff
@@ -763,3 +763,25 @@ export function unregisterToken( name ) {
 		name,
 	};
 }
+
+/**
+ * Returns an action object used in signalling that the user has enabled the publish sidebar.
+ *
+ * @return {Object} Action object
+ */
+export function enablePublishSidebar() {
```

Comment:
I originally added the enable/disable logic to the `core/edit-post` store and moved it later to the `core/editor`. See 28262d227182646cc0c5bc85d69999f16e5df2e4 The rationale for this change is that the PostPublishPanel (which lives in `@wordpress/editor`) uses this logic, and the `editor` can't depend on any component declared in `edit-post`.

An alternative approach could have been to keep the logic in `edit-post` and use the PrePublish and PostPublish extensions from `edit-post` to inject the checkbox component into the Publish sidebar. This is why I've discarded this approach:

- In my view, the extensions are meant for allowing plugins to inject their own panels into the PublishPanel. Dismissing the panel is a core action and using the extensions to circumvent the editor/edit-post separation would be a convoluted way to add support for it. To me, that'd be a smell that the editor/edit-post separation needs some more thinking.
- We want the checkbox to be shown in the same position of the panel (the bottom). To achieve this, we'd need to rely on CSS styling, because a plugin can't control other plugins's positions in the panel - which is prone to conflicts and unreliable positioning the checkbox.

Reply:
Is there another alternative to consider where the publish panel makes more sense in `edit-post` and could be moved wholesale there? This doesn't really feel like a component which ought be offered by the `editor` module.

Reply:
I wish we'd taken a more generalized approach akin to `edit-post`'s `toggleFeature` than to have specific action creators, selectors, reducer handling for each and every user preference.